reviewing/dependencies-hierarchy/test/index.ts

```diff
@@ -520,3 +521,46 @@ test('dependency without a package.json', async () => {
   expect(tree[withNonPackageDepFixture].dependencies![0]['devDependencies']).toBeUndefined()
   expect(tree[withNonPackageDepFixture].dependencies![0]['optionalDependencies']).toBeUndefined()
 })
+
+// Test case for https://github.com/pnpm/pnpm/issues/4814
+test('dependency tree shows correct depth even if previously visited with different depth', async () => {
```

Comment:
I'd like to add more test cases, but it's a bit difficult since testing different trees require setting up a distinct directory with `pnpm-lock.yaml`, and test go through `buildDependenciesHierarchy`.

Is there a way to export `getTree` for test visibility without exposing it on `dependencies-hierarchy`'s public exports?

Reply:
I guess you can write a low level test.

Reply:
Starting this in a separate PR https://github.com/pnpm/pnpm/pull/5828. If that looks reasonable I'll rebase this PR on top.

Reply:
Done! Latest PR adds more tests with the lower level tests set up in the prior merged PR above.